sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java

```diff
@@ -100,84 +111,72 @@ public List<BodyInterceptor> getBodyInterceptors() {
   @Nonnull
   public static PathBasedAnalysisInputLocation create(
       @Nonnull Path path, @Nonnull SourceType sourceType) {
-    return PathBasedAnalysisInputLocation.create(path, sourceType, Collections.emptyList());
+    return create(path, sourceType, Collections.emptyList());
   }
 
   @Nonnull
   public static PathBasedAnalysisInputLocation create(
       @Nonnull Path path,
       @Nonnull SourceType srcType,
       @Nonnull List<BodyInterceptor> bodyInterceptors) {
-    final PathBasedAnalysisInputLocation inputLocation;
+    return create(path, srcType, bodyInterceptors, Collections.emptyList());
+  }
+
+  @Nonnull
+  public static PathBasedAnalysisInputLocation create(
+      @Nonnull Path path,
+      @Nonnull SourceType srcType,
+      @Nonnull List<BodyInterceptor> bodyInterceptors,
+      @Nonnull Collection<Path> ignoredPaths) {
+
+    if (ignoredPaths.stream()
+        .anyMatch(ignoPath -> path.toString().startsWith(ignoPath.toString()))) {
+      throw new IllegalArgumentException(
```

Comment:
Shouldn't ignored input locations just not be considered for analysis?

Reply:
yes, but this way we can cut the loading earlier - especially it creates problems e.g. MultiReleaseJars would point to the jars root - which contains META-INF/versions/ as a subfolder and therefore would translate the complete path as the package name (and would resolve files it shouldnt); as a sideeffect this enables more precise "scoping" (again) as we removed the DefaultSourceTypeSpecifier